### ISSUE TITLE
Encoder applies no positional encoding; paper specifies 2D sine-cosine

### ISSUE THREAD

**Issue by engichang1467** [engichang1467/Open-Rig3R#42]
## Problem

The encoder currently applies no positional encoding at all. The paper specifies 2D
sine-cosine.

§3.3, Image Encoder:

> Rig3R employs a shared ViT-Large encoder to independently patchify and encode each input
> image using **2D sine-cosine positional encodings**. We initialize from DUSt3R.

`models/encoder_vit.py` builds the timm ViT with `pos_embed="none"`, so patch tokens carry
no spatial information into the encoder blocks.

## History

This is the tail of #34. Before that fix, the encoder *intended* sinusoidal encoding but
never achieved it, for two independent reasons:

```python
num_patches = self.vit.dump_patches          # nn.Module debug flag -> False
self.vit.pos_embed = nn.Parameter(
    self._build_sinusoidal_pos_embed(num_patches, embed_dim), requires_grad=False
)
```

- `dump_patches` is `nn.Module`'s debug flag, always `False`, so `num_patches == 0` and
  the resulting embedding was `(1, 1, C)` of **zeros**.
- torchvision's `VisionTransformer` has no `pos_embed` attribute — the real one is
  `vit.encoder.pos_embedding`, which `Encoder.forward` adds unconditionally. So the model
  was silently training on a **learned** embedding, not the frozen sinusoidal one the code
  claimed.

#34 replaced the torchvision backbone with timm to load the DUSt3R weights correctly, and
set `pos_embed="none"` rather than carry the broken code forward. The `ponytail:` comment
left at `models/encoder_vit.py:41` suggests porting CroCo's RoPE, on the reasoning that
the DUSt3R checkpoint uses `pos_embed='RoPE100'`. **That comment is wrong** — the
checkpoint uses RoPE, but Rig3R does not: §3.3 specifies 2D sine-cosine. The comment
should be corrected along with the fix.

## Why it matters

With `pos_embed="none"` the encoder is permutation-equivariant over patches: shuffling the
patches of an image produces the same shuffled tokens. Every downstream task here is
geometric. The DUSt3R weights being loaded were trained under RoPE, so they are also being
applied under a positional scheme they never saw.

## Suggested fix

Build a 2D sine-cosine table over the patch grid and add it to the tokens. The deleted
`_build_sinusoidal_pos_embed` in the #34 diff is a reasonable starting point — the encoding
scheme was fine, only `num_patches` and the attachment point were wrong. It needs:

- `num_patches` from `(img_size // patch_size) ** 2`, not `dump_patches`
- no `[CLS]` row (the timm encoder is built with `class_token=False`)
- attachment via timm's `pos_embed` parameter so `forward_features` actually applies it

Worth a check that the embedding is non-zero and that shuffling patches changes the
output — the original bug was invisible precisely because nothing asserted either.
